### PR TITLE
[WIP] Update package structure

### PR DESCRIPTION
**Pull request by orionarcher**
This PR implements several changes to the package structure that start to implement some SWE best practices. @jmythms 

Namely:
- testing structure
- documentation
- continuous integration actions
- packaging

A few points to discuss before merging this in:
- what can/should be tested?
- where should data files be stored and what python files need to be updated?

### CODE REVIEW

File: .lgtm.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Configure LGTM for this package
+
+extraction:
+  python:  # Configure Python
+    python_setup:  # Configure the setup
```
Comment (orionarcher) [inline .lgtm.yml:5]: this is trash, do better orion

Comment (jmythms) [reply .lgtm.yml:5]: Not trash 😆

**Comment by jmythms** [comment]
Our (very) basic tests passed. I have tested this version locally. Everything works. Merging in.

**Comment by jmythms** [comment]
Many thanks @orionarcher for your contribution!

**Comment by orionarcher** [comment]
Excellent job! Very happy to see it!

**Pull request merged**